vnext/Microsoft.ReactNative/JsiApi.idl

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.ReactNative
+{
+  interface IJsiRuntime;
+
+  delegate void JsiDataHandler(UInt8[] data);
+
+  [webhosthidden]
+  interface IJsiBuffer
+  {
+    UInt32 Size { get; };
+    void GetData(JsiDataHandler handler);
+  };
+
+  enum JsiValueType
+  {
+    Undefined,
+    Null,
+    Boolean,
+    Number,
+    Symbol,
+    String,
+    Object,
+  };
+
+  [webhosthidden]
+  struct JsiValueData
+  {
+    JsiValueType Kind;
+    UInt64 Value;
+  };
+
+  [default_interface]
+  runtimeclass JsiPreparedJavaScript
+  {
+  };
+
+  delegate JsiValueData JsiHostFunction(IJsiRuntime runtime, JsiValueData thisValue, JsiValueData[] args);
+
+  [webhosthidden]
+  interface IJsiHostObject
+  {
+    JsiValueData GetProperty(IJsiRuntime runtime, UInt64 name);
+    void SetProperty(IJsiRuntime runtime, UInt64 name, JsiValueData value);
+    IVectorView<UInt64> GetPropertyNames(IJsiRuntime runtime);
+  };
+
+  [webhosthidden]
+  interface IJsiRuntime
+  {
+    JsiValueData EvaluateJavaScript(IJsiBuffer buffer, String sourceUrl);
+    JsiPreparedJavaScript PrepareJavaScript(IJsiBuffer buffer, String sourceUrl);
+    JsiValueData EvaluatePreparedJavaScript(JsiPreparedJavaScript js);
+    UInt64 Global { get; };
+    String Description { get; };
+    Boolean IsInspectable { get; };
+
+    UInt64 CloneSymbol(UInt64 symbol);
+    UInt64 CloneString(UInt64 str);
+    UInt64 CloneObject(UInt64 obj);
+    UInt64 ClonePropertyNameId(UInt64 propertyNameId);
+
+    UInt64 CreatePropertyNameIdFromAscii(UInt8[] ascii);
+    UInt64 CreatePropertyNameIdFromUtf8(UInt8[] utf8);
+    UInt64 CreatePropertyNameIdFromString(UInt64 str);
+    void PropertyNameIdToUtf8(UInt64 propertyNameId, JsiDataHandler utf8Handler);
+    Boolean ComparePropertyNameIds(UInt64 left, UInt64 right);
+
+    void SymbolToUtf8(UInt64 symbol, JsiDataHandler utf8Handler);
+
+    UInt64 CreateStringFromAscii(UInt8[] ascii);
+    UInt64 CreateStringFromUtf8(UInt8[] utf8);
+    void StringToUtf8(UInt64 str, JsiDataHandler utf8Handler);
+
+    JsiValueData CreateValueFromJsonUtf8(UInt8[] json);
+
+    UInt64 CreateObject();
+    UInt64 CreateObjectWithHostObject(IJsiHostObject hostObject);
+    IJsiHostObject GetHostObject(UInt64 obj);
+    JsiHostFunction GetHostFunction(UInt64 func);
+
+    JsiValueData GetProperty(UInt64 obj, UInt64 propertyNameId);
+    JsiValueData GetPropertyWithString(UInt64 obj, UInt64 name);
```

Comment:
>GetPropertyWithString [](start = 17, length = 21)

How is that different from the previous method? Could you perhaps add a comment (perhaps containing a JS snippet representing the equivalent operation)?

Reply:
the facebook jsi provides two way to access property, one by property id, and another by string.

Reply:
I have changed the names and types in the iteration 6: the difference is that in one case we use property id as a symbol, and in the other we use use property name as a string.

---
In reply to: [451851341](https://github.com/microsoft/react-native-windows/pull/5461#discussion_r451851341) [](ancestors = 451851341)

Reply:
I agree, it seems to me quite useless too: The JSI implementation of JSC engine actually has the same implementation for both cases. Maybe we should consider to change the JSI ABI and use only one method.

---
In reply to: [452518028](https://github.com/microsoft/react-native-windows/pull/5461#discussion_r452518028) [](ancestors = 452518028)